components/theia/packages/gitpod-extension/src/browser/gitpod-user-storage-provider.ts

```diff
@@ -93,6 +94,37 @@ export class GitpodUserStorageProvider implements FileSystemProviderWithFileRead
         }]);
     }
 
+    // To avoid having badly written extensions executing too much requests and putting pressure on other components we
+    // cache writes here.
+    protected readonly contentCache = new Map<string, string>();
+    protected readonly updatedSinceLastWrite: string[] = [];
+    protected async cachedWrite(resource: GitpodServer.UpdateUserStorageResourceOptions): Promise<void> {
+        this.contentCache.set(resource.uri, resource.content);
+        this.updatedSinceLastWrite.push(resource.uri);
```

Comment:
probably it is evening and i'm missing something 😴 but if there are several calls to some resource only the last should win, we should drop the rest.

Reply:
Ah, thx! That's actually a :bug: Updates should be a map.